src/components/taskDetails/index.tsx

```diff
@@ -276,13 +278,17 @@ const TaskDetails: FC<Props> = ({ taskID }) => {
                         </section>
 
                         <section className={classNames.rightContainer}>
-                            <button
-                                onClick={() =>
-                                    router.push(`/progress/${taskID}?dev=true`)
-                                }
-                            >
-                                Update Progress
-                            </button>
+                            {isDevModeEnabled && (
+                                <button
+                                    onClick={() =>
+                                        router.push(
+                                            `/progress/${taskID}?dev=true`
+                                        )
+                                    }
+                                >
+                                    Update Progress
+                                </button>
+                            )}
```

Comment:
Based on the attached coverage, it appears that these lines have not been included in the tests. I kindly request your attention to ensure that adequate coverage is provided for these specific lines. It is important to address this matter to maintain the overall test comprehensiveness and effectiveness.